docs/lesiaturuta/PokemonProject/scripts/pokemon.js

```diff
@@ -0,0 +1,209 @@
+    cardWeightEl.textContent = `${pokemon.weight} lbs`;
+    cardNumberEl.textContent = pokemon.number;
+    const [firstType, secondType] = pokemon.type;
+    cardType1El.textContent = firstType;
+    cardType2El.textContent = secondType;
+    if (!secondType) {
+        cardType2El.classList.add('display-none');
+    }
+    const [firstWeakness, secondWeakness] = pokemon.weakness;
+    cardWeaknesses1El.textContent = firstWeakness;
+    cardWeaknesses2El.textContent = secondWeakness;
+    if (!secondWeakness) {
+        cardWeaknesses2El.classList.add('display-none');
+    }
+    return templateContent;
+};
+
+const cleanElement = (el) => {
+    while (el.firstChild) {
+        el.removeChild(el.firstChild);
+    }
+};
+
+const checkScroll = () => {
+    if (
+        document.body.scrollHeight > window.innerHeight
+        || document.documentElement.scrollHeight > window.innerHeight
+    ) {
+        document.body.style.height = '100%';
+    } else {
+        document.body.style.height = '100vh';
+    }
+};
+
+const renderCards = (cardContainer, pokemonsArray) => {
+    cleanElement(cardsEl);
+    const fragment = new DocumentFragment();
+    for (let i = 0; i < pokemonsArray.length; i++) {
+        fragment.append(createCardElement(pokemonsArray[i]));
+    }
+    cardContainer.append(fragment);
+    checkScroll();
+    return true;
+};
+
+const renderSelect = (name) => {
+    cleanElement(searchEl);
+    const optionFirstEl = document.createElement('option');
+    optionFirstEl.value = '';
+    optionFirstEl.text = 'All types';
+    searchEl.add(optionFirstEl);
+    typesPokemons.forEach((type) => {
+        const optionEl = document.createElement('option');
+        optionEl.value = type;
+        if (type === name) {
+            optionEl.selected = true;
+        }
+        optionEl.text = type;
+        searchEl.add(optionEl);
+    });
+};
+
+const sortPokemonsUp = (arrayPokemons) => arrayPokemons.sort((a, b) => a.height - b.height);
+
+const sortPokemonsDown = (arrayPokemons) => arrayPokemons.sort((a, b) => b.height - a.height);
+
+const getPokemonTypes = (arrayPokemons) => {
+    const types = [];
+    arrayPokemons.forEach((pokemon) => pokemon.type.forEach((type) => {
+        if (!types.includes(type)) types.push(type);
+    }));
+    return types;
+};
+
+const getPokemonsByType = (arrayPokemons) => (typeValue
+    ? arrayPokemons
+        .filter((pokemon) => pokemon.type
+            .includes(typeValue))
+    : arrayPokemons);
+
+const getPokemonsByName = (arrayPokemons) => (searchValue
+    ? arrayPokemons
+        .filter((pokemon) => pokemon.name
+            .toLocaleLowerCase()
+            .includes(searchValue.toLocaleLowerCase()))
+    : arrayPokemons);
+
+const getPokemonsByWeight = (arrayPokemons) => {
+    let newPokemons = [...arrayPokemons];
+    if (checkboxBig && checkboxSmall && newPokemons.length) {
+        newPokemons = newPokemons.filter((pokemon) => pokemon.weight < 100 && pokemon.weight > 50);
```

Comment:
your logic is broken
<img width="867" alt="image" src="https://github.com/MastersAcademy/fe-course-basic-2023/assets/98031768/14fab857-a135-4af6-93e9-db8eb2c137df">

Reply:
fixed